### PR TITLE
2069 - parse to csv stream

### PR DESCRIPTION
**Pull request by islean**
## Description

Issue 2069, echo nested list as a csv stream when creating a sample sheet with a dry run. 

### Fixed

- Now prints nested list as a csv stream

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b 2069-csv-parsing -a
    ```

### How to test

- [ ] Do a dry run using the CLI and check that the correct formatting is applied

### Expected test outcome

- [ ] Check that the printed file seems like a csv file
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/demultiplex/sample_sheet.py [hunk 1]
```diff
@@ -102,6 +101,10 @@ def create_sheet(
     )
 
 
+def _list_to_csv_stream(nested_list: List[List[str]]) -> str:
+    return WriteStream().write_stream_from_content(file_format=FileFormat.CSV, content=nested_list)
```
Comment (henrikstranneheim) [inline cg/cli/demultiplex/sample_sheet.py:105]:
```suggestion
    return WriteStream.write_stream_from_content(file_format=FileFormat.CSV, content=nested_list)
```
Do we need this extra utility function? I think you can call Writestream directly in the click echo

**Review by henrikstranneheim — COMMENTED** [review]
👍

**Comment by Vince-janv** [comment]
### Output is correctly formatted
```
cg demultiplex samplesheet create 221019_A00689_0663_BH7FYHDRX2 --dry-run
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Running cg demultiplex.
Creating sample sheet for flowcell 221019_A00689_0663_BH7FYHDRX2
Set environment to stage
Fetching samples from lims for flowcell H7FYHDRX2
Constructing sample sheet for H7FYHDRX2
Skip adding dummy samples since run is not WGS
Removing all samples without dual indexes
Fix so that all indexes are on the correct format
Check if run is reverse complement
Converting reagent kit parameter 3 to version
Run is reverse complement
Create sample sheet for samples
Validating sample sheet
Order samples by lane
Validate that samples are unique in lane 1
Validate that samples are unique in lane 2
Sample sheet passed validation
[Settings]
BarcodeMismatchesIndex1,0
BarcodeMismatchesIndex2,0
[Data]
FCID,Lane,SampleID,SampleRef,index,index2,SampleName,Control,Recipe,Operator,Project
<fc-name>,<sample-id>,hg19,GATCAACA,TAACATAG,<sample-name>,N,R1,script,894614
```

File: cg/cli/demultiplex/sample_sheet.py (hunk 1 above)
Comment (islean) [reply cg/cli/demultiplex/sample_sheet.py]: Reasonable, changed now.

**Review by henrikstranneheim — APPROVED** [review]
Great 👍

**Review by diitaz93 — APPROVED** [review]
Looks great 💯

**Comment by islean** [comment]
Deployed on 2023-05-29 by IO and VJ

**Pull request merged**